### PR TITLE
Remove datasets for foundation models online inference samples

### PR DESCRIPTION
**Pull request by susridhar**
# Description

This PR contains updated online inference samples for foundation models that download the datasets themselves instead of relying on data checked in to the repo.

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/Azure/azureml-examples/blob/main/CONTRIBUTING.md)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by ManojBableshwar — APPROVED** [review]

File: sdk/python/foundation-models/system/inference/automatic-speech-recognition/asr-online-endpoint.ipynb [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+   "source": [
+    "## Automatic Speech Recognition Inference using Online Endpoints\n",
+    "\n",
+    "This sample shows how to deploy `automatic-speech-recognition` type models to an online endpoint for inference.\n",
+    "\n",
+    "### Task\n",
+    "`automatic-speech-recognition` (ASR) converts a speech signal to text, mapping a sequence of audio inputs to text outputs. Virtual assistants like Siri and Alexa use ASR models to help users everyday, and there are many other useful user-facing applications like live captioning and note-taking during meetings.\n",
+    "\n",
+    "### Model\n",
+    "Models that can perform the `automatic-speech-recognition` task are tagged with `task: automatic-speech-recognition`. We will use the `openai-whisper-large` model in this notebook. If you opened this notebook from a specific model card, remember to replace the specific model name. If you don't find a model that suits your scenario or domain, you can discover and [import models from HuggingFace hub](../../import/import-model-from-huggingface.ipynb) and then use them for inference. \n",
```
Comment (amymsft) [inline sdk/python/foundation-models/system/inference/automatic-speech-recognition/asr-online-endpoint.ipynb:16]: The import links are broken on all the notebooks. Can you update all to (../../import/import_model_into_registry.ipynb)

File: sdk/python/foundation-models/system/inference/fill-mask/book-corpus-dataset/download-dataset.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+args = parser.parse_args()
+
+# create the download directory if it does not exist
+if not os.path.exists(args.download_dir):
+    os.makedirs(args.download_dir)
+
+# import hugging face datasets library
+from datasets import load_dataset, get_dataset_split_names
+
+for split in get_dataset_split_names(args.dataset):
```
Comment (amymsft) [inline sdk/python/foundation-models/system/inference/fill-mask/book-corpus-dataset/download-dataset.py:23]: Do you need it for every split? Why not just "train"?

File: cli/foundation-models/system/inference/text-classification/text-classification-online-endpoint.sh [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+# the sample scoring file available in the same folder as the above notebook
+
+# script inputs
+registry_name="azureml"
+subscription_id="<SUBSCRIPTION_ID>"
+resource_group_name="<RESOURCE_GROUP>"
+workspace_name="<WORKSPACE_NAME>"
+
+# This is the model from system registry that needs to be deployed
+model_name="microsoft-deberta-base-mnli"
```
Comment (amymsft) [inline cli/foundation-models/system/inference/text-classification/text-classification-online-endpoint.sh:12]: This model is from the entailment/contraction text classification type.  Update this model to match the sentiment analysis one from SDK.

**Review by amymsft — CHANGES_REQUESTED** [review]
Few suggestions. I think the CLI for text classification will not work with the current model and sample input though.

**Comment by amymsft** [comment]
For smoke test failure, Haritha showed me how to use the black formatter. 


pip install black black[jupyter] and run  black <directory name>

File: sdk/python/foundation-models/system/inference/automatic-speech-recognition/asr-online-endpoint.ipynb (hunk 1 above)
Comment (susridhar) [reply sdk/python/foundation-models/system/inference/automatic-speech-recognition/asr-online-endpoint.ipynb:16]: Fixed

File: sdk/python/foundation-models/system/inference/fill-mask/book-corpus-dataset/download-dataset.py (hunk 2 above)
Comment (susridhar) [reply sdk/python/foundation-models/system/inference/fill-mask/book-corpus-dataset/download-dataset.py:23]: Updated to just download "train"

File: sdk/python/foundation-models/system/inference/summarization/summarization-online-endpoint.ipynb [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+    "# read the ./news-summary-dataset/train.jsonl file into a pandas dataframe\n",
+    "df = pd.read_json(\"./news-summary-dataset/train.jsonl\", lines=True)\n",
+    "# escape single and double quotes in the masked_text column\n",
+    "df[\"article\"] = df[\"article\"].str.replace(\"'\", \"\\\\'\").str.replace('\"', '\\\\\"')\n",
+    "# pick 1 random row\n",
+    "sample_df = df.sample(1)\n",
+    "# create a json object with the key as \"inputs\" and value as a list of values from the article column of the sample_df dataframe\n",
+    "sample_json = {\"inputs\": sample_df[\"article\"].tolist()}\n",
+    "# save the json object to a file named sample_score.json in the ./news-summary-dataset folder\n",
+    "test_json = {\"inputs\": {\"input_string\": sample_df[\"article\"].tolist()}}\n",
```
[the remark this answers is not in the document]
Comment (susridhar) [reply sdk/python/foundation-models/system/inference/summarization/summarization-online-endpoint.ipynb:230]: Long inputs will simply cause a timeout, which the user will be able to see.

File: cli/foundation-models/system/inference/text-classification/text-classification-online-endpoint.sh (hunk 3 above)
Comment (susridhar) [reply cli/foundation-models/system/inference/text-classification/text-classification-online-endpoint.sh:12]: Fixed

File: sdk/python/foundation-models/system/inference/text-classification/text-classification-online-endpoint.ipynb [hunk 5]
```diff
@@ -0,0 +1,327 @@
...
+    "This sample shows how to deploy `text-classification` type models to an online endpoint for inference.\n",
+    "\n",
+    "### Task\n",
+    "`text-classification` is generic task type that can be used for scenarios such as sentiment analysis, emotion detection, grammar checking, spam filtering, etc. In this example, we will test for entailment v/s contradiction, meaning given a premise sentence and a hypothesis sentence, the task is to predict whether the premise entails the hypothesis (entailment), contradicts the hypothesis (contradiction), or neither (neutral). \n",
+    "\n",
+    "### Inference data\n",
+    "We will use the [imdb](https://huggingface.co/datasets/imdb) dataset\n",
+    "\n",
+    "### Model\n",
+    "Look for models tagged with `text-classification` in the system registry. Just looking for `text-classification` is not sufficient, you need to check if the model is specifically finetuned for  entailment v/s contradiction by studying the model card and looking at the input/output samples or signatures of the model. In this notebook, we use the `finiteautomata-bertweet-base-sentiment-analysis` model.\n",
```
[the remark this answers is not in the document]
Comment (susridhar) [reply sdk/python/foundation-models/system/inference/text-classification/text-classification-online-endpoint.ipynb:19]: Done

**Review by amymsft — APPROVED** [review]

**Pull request merged**